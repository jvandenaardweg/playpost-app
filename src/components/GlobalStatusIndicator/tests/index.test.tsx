import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';

import { GlobalStatusIndicator, Props } from '../index';

const defaultProps: Props = {
  audiofileStatus: '',
  playlistIsLoadingCreateItem: false
}

describe('GlobalStatusIndicator', () => {
  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeAll(() => {
      const props = {
        ...defaultProps,
      }
      wrapper = render(<GlobalStatusIndicator {...props} />);
    });

    it('should render "Creating article audio..."', () => {
      const props = {
        ...defaultProps,
        audiofileStatus: 'Creating article audio...',
        playlistIsLoadingCreateItem: false
      }

      wrapper.update(<GlobalStatusIndicator {...props} />);

      expect(wrapper.getByTestId('GlobalStatusIndicator-label').props.children).toBe('Creating article audio...');
    });

    it('should render "Adding article..."', () => {
      const props = {
        ...defaultProps,
        playlistIsLoadingCreateItem: true
      }

      wrapper.update(<GlobalStatusIndicator {...props} />);

      expect(wrapper.getByTestId('GlobalStatusIndicator-label').props.children).toBe('Adding article...');
    });

    it('should use a fallback label when label is not set', () => {
      const props = {
        ...defaultProps,
        playlistIsLoadingCreateItem: false
      }

      wrapper.rerender(<GlobalStatusIndicator {...props} />);

      expect(wrapper.getByTestId('GlobalStatusIndicator-label').props.children).toBe('Loading...');
    });
  });
});
