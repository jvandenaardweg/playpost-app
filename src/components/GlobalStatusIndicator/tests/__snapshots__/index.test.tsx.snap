// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStatusIndicator rendering should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": 50,
      "justifyContent": "center",
      "position": "absolute",
      "top": -56,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#037DE2",
        "borderRadius": 100,
        "flexDirection": "row",
        "height": 38,
        "justifyContent": "center",
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="white"
      hidesWhenStopped={true}
      size="small"
    />
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 15,
          "fontWeight": "600",
          "marginLeft": 10,
        }
      }
      testID="GlobalStatusIndicator-label"
    >
      Creating article audio...
    </Text>
  </View>
</View>
`;
